x-pack/plugins/index_lifecycle_management/public/application/services/policies/warm_phase.ts

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { isEmpty } from 'lodash';
+import {
+  AllocateAction,
+  WarmPhase,
+  SerializedWarmPhase,
+  serializedPhaseInitialization,
+} from './types';
+import { isNumber, splitSizeAndUnits } from './policy_serialization';
+
+import {
+  numberRequiredMessage,
+  PhaseValidationErrors,
+  positiveNumberRequiredMessage,
+  positiveNumbersAboveZeroErrorMessage,
+} from './policy_validation';
+
+const warmPhaseInitialization: WarmPhase = {
+  phaseEnabled: false,
+  warmPhaseOnRollover: false,
+  selectedMinimumAge: '0',
+  selectedMinimumAgeUnits: 'd',
+  selectedNodeAttrs: '',
+  selectedReplicaCount: '',
+  shrinkEnabled: false,
+  selectedPrimaryShardCount: '',
+  forceMergeEnabled: false,
+  selectedForceMergeSegments: '',
+  phaseIndexPriority: '',
+};
+
+export const warmPhaseFromES = (phaseSerializedOrUndefined?: SerializedWarmPhase): WarmPhase => {
+  const phase: WarmPhase = { ...warmPhaseInitialization };
+
+  if (!phaseSerializedOrUndefined) {
+    return phase;
+  }
+
+  // after ! check it's safe to cast type as a serialized phase
+  const phaseSerialized = phaseSerializedOrUndefined as SerializedWarmPhase;
```

Comment:
Same here about naming + casting. I see that we are already doing the check on L40, so the casting should not be required, and we shouldn't need a different variable for it. (`phaseSerialized` could be the name of the prop)